pandas/core/reshape/merge.py

```diff
@@ -1138,6 +1141,88 @@ def _get_join_indexers(left_keys, right_keys, sort=False, how='inner',
     return join_func(lkey, rkey, count, **kwargs)
 
 
+def _restore_dropped_levels_multijoin(left, right, dropped_level_names,
+                                      join_idx, lidx, ridx):
+    """
+    *this is an internal non-public method*
+
+    Returns the levels, labels and names of a multil-index to multi-index join.
+    Depending on the type of join, this method restores the appropriate
+    dropped levels of the joined multi-index. The method relies on lidx, ridx
+    which hold the index positions of left and right, where a join was feasible
+
+    Parameters
+    ----------
+    left : Index
+        left index
+    right : Index
+        right index
+    dropped_level_names : str array
+        list of non-common levels
+    join_idx : Index
+        the index of the join between the common levels of left and right
+    lidx : intp array
+        left indexer
+    right : intp array
+        right indexer
+
+    Returns
+    -------
+    levels : intp ndarray
+        levels of combined multiindexes
+    labels : str array
+        labels of combined multiindexes
+    names : str array
+        names of combined multiindexes
+
+    """
+
+    # Convert to 1 level multi-index if not
+    if not isinstance(join_idx, MultiIndex):
+        levels = [join_idx.values]
+        labels = [list(range(0, len(join_idx)))]
```

Comment:
Style nit: all these `range(0, len(.))` can just be `range(.)`.

Reply:
This approach gave me errors but probably I'm missing something.
I instead used [range(join_idx.size)] assuming that it may be faster than [range(len(join_idx))]